Gateway/Http/TransactionBuilder/Idin.php

```diff
@@ -0,0 +1,193 @@
+ * versions in the future. If you wish to customize this module for your
+ * needs please contact support@buckaroo.nl for more information.
+ *
+ * @copyright Copyright (c) Buckaroo B.V.
+ * @license   https://tldrlegal.com/license/mit-license
+ */
+
+namespace Buckaroo\Magento2\Gateway\Http\TransactionBuilder;
+
+use Buckaroo\Magento2\Logging\Log;
+use Magento\Framework\UrlInterface;
+use Magento\Framework\Data\Form\FormKey;
+use Magento\Framework\Encryption\Encryptor;
+use Magento\Store\Model\StoreManagerInterface;
+use Buckaroo\Magento2\Gateway\Http\Transaction;
+use Buckaroo\Magento2\Model\ConfigProvider\Account;
+use Magento\Customer\Model\Session as CustomerSession;
+use Magento\Framework\App\Config\ScopeConfigInterface;
+use Buckaroo\Magento2\Model\ConfigProvider\Method\Factory;
+use Buckaroo\Magento2\Service\Software\Data as SoftwareData;
+
+class Idin extends AbstractTransactionBuilder implements IdinBuilderInterface
+{
+
+    /**
+     * @var \Magento\Store\Api\Data\StoreInterface
+     */
+    protected $store;
+
+    /**
+     * @var mixed
+     */
+    protected $customerId;
+
+    /**
+     *
+     * @var string
+     */
+    protected $issuer;
+
+    /** @var Encryptor $encryptor */
+    private $encryptor;
+
+    /**
+     * @var Account
+     */
+    protected $configProviderAccount;
+
+    /**
+     * @param ScopeConfigInterface $scopeConfig
+     * @param SoftwareData $softwareData
+     * @param Account $configProviderAccount
+     * @param Transaction $transaction
+     * @param UrlInterface $urlBuilder
+     * @param FormKey $formKey
+     * @param Encryptor $encryptor
+     * @param Factory $configProviderMethodFactory
+     * @param Log $logging
+     */
+    public function __construct(
+        ScopeConfigInterface $scopeConfig,
+        SoftwareData $softwareData,
+        Account $configProviderAccount,
+        Transaction $transaction,
+        UrlInterface $urlBuilder,
+        FormKey $formKey,
+        Encryptor $encryptor,
+        Factory $configProviderMethodFactory,
+        Log $logging,
+        StoreManagerInterface $storeManager,
+        CustomerSession $customerSession
+    ) {
+        $this->store = $storeManager->getStore();
+        $this->customerId =  $customerSession->getCustomerId();
+        $this->encryptor = $encryptor;
+        $this->configProviderAccount = $configProviderAccount;
+
+        parent::__construct(
+            $scopeConfig,
+            $softwareData,
+            $configProviderAccount,
+            $transaction,
+            $urlBuilder,
+            $formKey,
+            $encryptor,
+            $configProviderMethodFactory,
+            $logging
+        );
+    }
+
+    /**
+     * Set issuer 
+     *
+     * @param string $issuer
+     *
+     * @return $this
+     */
+    public function setIssuer(string $issuer)
+    {
+        $this->issuer = $issuer;
+        return $this;
+    }
+    /**
+     * @return array
+     */
+    public function getBody()
```

Comment:
does this apply only to idin or will it be used by other methods (in order to identify if it needs a generic/std approach for all or it's just for this one).

Reply:
this builder is specifically for idin, we currently don't have other age verification methods so i think this should remain how it is now, previously the `datarequest`  builder was responsible with the creation of the transaction request but that builder requires a order to properly build the request, since idin is called before the creation of a order i removed this dependency